### PR TITLE
Rust UI: loaders, hold to confirm, pagination for model T

### PR DESCRIPTION
**Pull request by mmilata**
Followup of #1960.

### CODE REVIEW

**Review by mmilata — COMMENTED** [review]
Looks good, lots of stuff we're gonna need. Loader works smoothly. I propose to merge this after #1960, bugs will probably be found & fixed when we write some layouts using these components. Also let's change `chore` in commit messages to `fix`/`feat`, iirc chores are meant to have no source code changes.

Things I'd like to work on based on this PR:
- decouple `Paginated` from TT - factor out theme, write `Page` component for T1
- `impl Paginate for Paragraphs`

File: core/embed/rust/src/ui/model_tt/component/loader.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                    ctx.request_anim_frame();
+                }
+            }
+        }
+        None
+    }
+
+    fn paint(&mut self) {
+        // TODO: Should we paint based on the time in `Self::event`, or based on the
+        // current instant?
```
Comment (matejcik) [inline core/embed/rust/src/ui/model_tt/component/loader.rs:137]: current instant seems significantly more appropriate (time between generating the event and receiving it goes away)
also currently events don't propagate time, no?

**Comment by matejcik** [comment]
LGTM except the two comments

**Comment by matejcik** [comment]
squashed & rebased onto master (with #1960 merged)

I had to re-add `Mul<f32> for Duration`, at least now I know why it's safe, and added explanation in comments: 9c2387a

File: core/src/trezor/loop.py [hunk 2]
```diff
@@ -558,3 +558,16 @@ def is_running(self) -> bool:
...
+
+    def __init__(self) -> None:
+        self.task: Task | None = None
+
+    def handle(self, task: Task) -> None:
+        self.task = task
+
+    def schedule(self, deadline: int, value: Any) -> None:
+        if self.task is not None:
+            schedule(self.task, value)
```
[the remark this answers is not in the document]
Comment (matejcik) [reply core/src/trezor/loop.py:586]: 77804024

File: core/embed/rust/src/ui/model_tt/component/loader.rs (hunk 1 above)
Comment (matejcik) [reply core/embed/rust/src/ui/model_tt/component/loader.rs:137]: 67a4dc37

**Pull request merged**